src/language/ast.js

```diff
@@ -385,19 +402,38 @@ export type ListValueNode = {|
   +values: $ReadOnlyArray<ValueNode>,
 |};
 
+export type ConstListValueNode = {|
+  +kind: 'ListValue',
```

Comment:
I understand why we use `ListValue` in both types.
But it means we lose the ability to distinguish them in runtime.
Also, it makes `isConstValueNode` the only predicate that recursive check nodes.

Can we do:
```js
export type ConstListValueNode = {|
  +kind: 'ListValue',
  +loc?: Location,
  +values: $ReadOnlyArray<ConstValueNode>,
  +isConst: true,
|};

export type ListValueNode = ConstListValueNode | {|
  +kind: 'ListValue',
  +loc?: Location,
  +values: $ReadOnlyArray<ValueNode>,
  +isConst: false,
|}
```

@leebyron What do you think?

Reply:
I'd rather not add Booleans to encode this since it's already knowable from the structure itself, and this would add two sources of truth that could disagree and would need to be carefully kept in sync during any edits.

For example parsing a value with parseValue() instead of parseConstValue() returns ValueNode instead of ConstValueNode since that's the safe type, but it might be const if it doesn't contain a variable. 

That means the parser would need an inlined version of isConstValue to set the booleans correctly. 

Another example is editing one with visit() to add or remove Variables would need to update these booleans. Right now that's not necessary

Reply:
@leebyron Argument about changing totally makes sense 👍